From 8f0f82d573f8a2e40cd9355354694e5d20cf81e7 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Mon, 30 Sep 2024 14:31:40 +0200
Subject: [PATCH] dt-bindings: soc: st: add RISAB documentation

Add documentation on the RISAB peripheral that is a memory firewall.

Change-Id: Ia157a609175a99610666e41c077a6df76001351f
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/407779
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 .../bindings/soc/st/st,stm32mp25-risab.yaml   | 67 +++++++++++++++++++
 1 file changed, 67 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/soc/st/st,stm32mp25-risab.yaml

diff --git a/Documentation/devicetree/bindings/soc/st/st,stm32mp25-risab.yaml b/Documentation/devicetree/bindings/soc/st/st,stm32mp25-risab.yaml
new file mode 100644
index 000000000000..5febe1463580
--- /dev/null
+++ b/Documentation/devicetree/bindings/soc/st/st,stm32mp25-risab.yaml
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/soc/st/st,stm32mp25-risab.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32 Resource isolation peripheral unit for address space protection
+       (block-based)
+
+maintainers:
+  - Gatien Chevallier <gatien.chevallier@foss.st.com>
+
+description: |
+  The RIF (resource isolation framework) is a comprehensive set of hardware
+  blocks designed to enforce and manage isolation of STM32 hardware resources,
+  like memory and peripherals.
+  Through RISAB registers, a trusted domain, or the domain to whom the
+  page configuration has been delegated, assigns memory pages to one or more
+  security domains (secure, privilege, compartment).
+
+properties:
+  compatible:
+    const: st,stm32mp25-risab
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  st,mem-map:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    minItems: 2
+    description:
+      Defines the memory address range covered by the RISAB.
+
+  memory-region:
+    minItems: 1
+    maxItems: 32
+    description: |
+      Phandle to a node describing memory areas to be configured by the RISAB.
+      These regions cannot overlap. A zone must be within st,mem-map range and
+      can be represented by one or more pages.
+
+  st,srwiad:
+    description: |
+      When set, this bit allows secure read/write data accesses to non-secure
+      blocks and pages (secure execute remains illegal).
+    type: boolean
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - st,mem-map
+
+additionalProperties: false
+
+examples:
+  - |
+        risab1: risab@420f0000 {
+          compatible = "st,stm32mp25-risab";
+          reg = <0x420f0000 0x1000>;
+          clocks = <&rcc CK_ICN_LS_MCU>;
+          st,mem-map = <0xa000000 0x20000>;
+          st,srwiad;
+        };
-- 
2.39.5

